Client/Frontend/Browser/Playlist/PlaylistInfo.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+
+private let log = Logger.browserLogger
+
+struct PlaylistInfo: Codable {
+    let name: String
+    let src: String
+    let pageSrc: String
+    let pageTitle: String
+    let mimeType: String
+    let duration: Float
+    let detected: Bool
+    let dateAdded: Date
+    
+    init(item: PlaylistItem) {
+        self.name = item.name ?? ""
+        self.src = item.mediaSrc ?? ""
+        self.pageSrc = item.pageSrc ?? ""
+        self.pageTitle = item.pageTitle ?? ""
+        self.mimeType = item.mimeType ?? ""
```

Comment:
Would it not be better to make this init return `nil` if this data doesn't exist? (i.e. `init?(item: PlaylistItem) { ... }`)

Reply:
No it won't. In that case I would unwrap all values from the CoreData object manually. I added this initializer so I don't have to do that. All values in CoreData are still optional even if I specify that they shouldn't be. It generates the fields as optional still so I added this method instead to unwrap each field at once.